### ISSUE TITLE
Hyperlinks not working always

### ISSUE THREAD

**Issue by alosarjos** [alacritty/alacritty#8270]
### System

OS: Linux (Archlinux)
Version: alacritty 0.14.0 (22a44757)
Linux/BSD: Gnome 47.1 + Wayland

There is something wrong with hyperlinks. I have opened two terminals: Alacritty and Gnome Console. On Gnome Console, hyperlinks are working just fine, on Alacritty I'm not sure how but when trying copy pasting the prompt or some gibberish on it, then it starts working

https://github.com/user-attachments/assets/20b78598-4364-4106-9570-445d3c927f75

**Comment by chrisduerr** [comment]
Sounds like a shell issue. The shell is likely rewriting itself when you're pasting text in thus fixing the missing hyperlink escape.

**Comment by alosarjos** [comment]
But it's working fine on another terminal by default, we sure it's not an alacritty issue?

**Comment by nixpulvis** [comment]
It would help to see what the format of the escape sequence you're using to create the link.

Chris is probably right, and the reason it could be different with another terminal is because not all terminals handle prompts the same way.

**Comment by chrisduerr** [comment]
> Chris is probably right, and the reason it could be different with another terminal is because not all terminals handle prompts (reflowing text) the same way.

Startup time and feature detection might also impact this.

**Comment by alosarjos** [comment]
I've checked since I was pretty sure this was working at some point, looks like maybe some regression on 0.14? I downgraded to 0.13 and recorded this video


https://github.com/user-attachments/assets/aef38c33-22e9-4611-bfc6-5529a17c725f

**Comment by nixpulvis** [comment]
Does it work on 0.14 if you run `alacritty -V` first and repaint the prompt like in the second video?

**Comment by alosarjos** [comment]
> Does it work on 0.14 if you run `alacritty -V` first and repaint the prompt like in the second video?

I'm sorry, I dont get what you mean by "repaint"

**Comment by alosarjos** [comment]
If I'm guessing correctly what you mean, looks to be handling the hyperlinks fine on old prompts


https://github.com/user-attachments/assets/68884a3f-c894-46e3-8059-7cd7f3ac7b41

**Comment by alosarjos** [comment]
Is it then something on Alacritty or something I should report somewhere else?

**Comment by kchibisov** [comment]
You can always try bisecting if you feel like it's a regression, but it could just be timing issue.

**Comment by nixpulvis** [comment]
This does look like a bug in alacritty. I can reproduce it in bash with:

```sh
PS1='\e]8;;https://nixpulvis.com\anixpulvis\e]8;;\a $ '
```

The current prompt always fails to be a clickable URL though the cursor does change and flicker a bit. This makes me think it's not a timing issue.

**Comment by alosarjos** [comment]
Recloned and retried going to the same commit and parent and reproduced, (Trying to figure out if it's something about timing)


https://github.com/user-attachments/assets/d0b3e49c-07de-4c05-8041-dea0abbd1ab4

**Comment by nixpulvis** [comment]
I also see a1ed79bd2c014be49a85e2100ce374b86c8839e8 introducing this issue when I bisect. I don't think it's a timing or reflow issue since the bug appears in a short prompt (see my above comment) and it's always present for the current prompt. Other prompts work as expected.

**Comment by chrisduerr** [comment]
Very possible for this patch to have introduced a regression.

**Comment by alosarjos** [comment]
If it helps, ping me whenever you want me to test any patch ;)

**Comment by nixpulvis** [comment]
Here's a video showcasing the issue a bit more broadly:
![damage](https://github.com/user-attachments/assets/c372715a-e24b-4bed-982c-002ec247d4bb)

**Comment by fabian-thomas** [comment]
Can confirm that a1ed79bd2c014be49a85e2100ce374b86c8839e8 introduced the regression. Here is a simple reproducer (just a hyperlink with one-character label, everything bigger than one char still works fine):
```sh
printf '\e]8;id=1235;https://example.com\e\\A\e]8;;\e\\\n'
```

**Comment by chrisduerr** [comment]
Should be fixed with #8276, let me know if you can still reproduce this.
